File: archiso/airootfs/usr/share/churros/preferences/widgets/navigation_row.py
import gi

gi.require_version("Gtk", "4.0")

from gi.repository import Gtk

from widgets.row import Row


class NavigationRow(Row):

    def __init__(
        self,
        navigator,
        title,
        icon,
        page_name,
        subtitle=None
    ):

        self.navigator = navigator
        self.page_name = page_name

        arrow = Gtk.Image.new_from_icon_name(
            "go-next-symbolic"
        )

        arrow.add_css_class(
            "row-arrow"
        )

        super().__init__(

            title=title,

            subtitle=subtitle,

            icon=icon,

            suffix=arrow,

            callback=self.on_clicked

        )

    def on_clicked(
        self,
        row
    ):

        self.navigator.show_page(
            self.page_name
        )

File: rust/preferences/src/widgets/navigation_row.rs
// ==========================================
// NavigationRow — fila que navega a otra página
// (equivalente a widgets/navigation_row.py)
// ==========================================

use gtk::prelude::*;

use crate::widgets::row::Row;

pub fn new(
    navigator: gtk::Stack,
    title: &str,
    icon: &str,
    page_name: &str,
    subtitle: Option<&str>,
) -> Row {
    let arrow = gtk::Image::from_icon_name("go-next-symbolic");
    arrow.add_css_class("row-arrow");

    let arrow_widget: &gtk::Widget = arrow.upcast_ref();
    let stack = navigator.clone();
    let target = page_name.to_string();

    Row::new(
        title,
        subtitle,
        Some(icon),
        None,
        Some(arrow_widget),
        Some(Box::new(move |_| {
            stack.set_visible_child_name(&target);
        })),
    )
}
